test_requirements.txt

```diff
@@ -0,0 +1,6 @@
+psutil
+pyopenssl
+pysendfile; platform_system != "Windows"
+mock==1.0.1
+unittest2
+pypiwin32; platform_system == "Windows"
```

Comment:
please revert this; I don't like having extra files in the root directory

Reply:
@giampaolo would you have pointers on how to implement OS dependent `pip install` in a `Makefile`?

Reply:
> ok, so the Makefile will only support Windows in GHA context? 

The Makefile can only be used on UNIX. Any automation / cmdline work on Windows should be executed by `tests.yml` itself in form of... commands.

Reply:
there is no GHA BSD runner. let me add `macos`

https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources

Reply:
`macos` fails with https://github.com/giampaolo/pyftpdlib/actions/runs/3200992815/jobs/5228523820